powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/ActivationServiceBehavior.java

```diff
@@ -841,18 +843,14 @@ public InitActivationResponse initActivation(InitActivationRequest request) thro
                 timestampExpiration = new Date(timestamp.getTime() + powerAuthServiceConfiguration.getActivationValidityBeforeActive());
             }
 
-            // Validate combination of activation OTP and OTP validation mode.
-            final boolean hasActivationOtp = activationOtp != null && !activationOtp.isEmpty();
             if (activationOtpValidation == null) {
                 activationOtpValidation = com.wultra.security.powerauth.client.model.enumeration.ActivationOtpValidation.NONE;
             }
-            if ((activationOtpValidation == com.wultra.security.powerauth.client.model.enumeration.ActivationOtpValidation.NONE && hasActivationOtp) ||
-                    (activationOtpValidation != com.wultra.security.powerauth.client.model.enumeration.ActivationOtpValidation.NONE && !hasActivationOtp)) {
-                logger.warn("Activation OTP doesn't match its validation mode.");
-                throw localizationProvider.buildExceptionForCode(ServiceError.INVALID_REQUEST);
-            }
+
```

Comment:
And we remove the validation when activationOtpValidation is not NONE but the activation OTP is not provided.
No impact on functionality, but we can add validation, but this can clash with the conditions validation OTP later on.